sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -325,7 +325,7 @@ object CreateStruct extends FunctionBuilder {
    */
   val registryEntry: (String, (ExpressionInfo, FunctionBuilder)) = {
     val info: ExpressionInfo = new ExpressionInfo(
-      "org.apache.spark.sql.catalyst.expressions.NamedStruct",
+      getClass.getCanonicalName.stripSuffix("$"),
```

Comment:
hmm, is it actually a `CreateNamedStruct`?

Reply:
I guess because `CreateStruct.apply` is the actual `FunctionBuilder`? I think either way is fine. `CreateNamedStruct` sounds fine to me too. This class isn't used anywhere in Spark if I am not mistaken so won't be a big deal.